Improve event support impl doc (#75)
jenkinsci/scm-api-plugin: docs/implementation.adoc

This can be problematic where, for example, the Jenkins server is on an internal-only network and the source control system is an externally hosted service (e.g. GitHub)

* Messaging service

The messaging service uses a broker which can be reached by both the Jenkins server and the source control system.
The source control system sends its event payloads to the broker system.
The Jenkins server periodically connects (or in some cases uses a persistent connection) to the broker to receive the payloads.

The webhook technique is the simpler to implement and is generally sufficient for most Jenkins users.
For the users where the webhook technique is not sufficient it is usually relatively easy to build a generic messaging service on top of the webhook, for example the https://wiki.jenkins-ci.org/display/JENKINS/SCM+SQS+Plugin[SCM SQS Plugin].

=== Setting Up Webhook

The basic starting point for a WebHook is an `UnprotectedRootAction`

[source,java]
----
@Extension
public class MySCMWebHook implements UnprotectedRootAction {
    public static final String URL_NAME = "my-scm-hook";

>     public static final String ENPOINT = "notify";

Reviewer: ```suggestion
    public static final String ENDPOINT = "notify";
```

}
----

> Basically, you don't have to worry about triggering the build... you just have to fire the event. If you event is implemented correctly then the event subsystem will match the event to the corresponding SCMSource instances and let them do what they need to do (and in the case of multibranch, what it needs to do is: initiate a check of the latest revision and then build if the revision is changed).

Reviewer: Started with typo fix then added a bit more. 
```suggestion
You don't have to worry about triggering the build, you just have to fire the event. If your event is implemented correctly then the event subsystem will match the event to the corresponding SCMSource instances and let them do what they need to do. For example, in the case of multibranch, they will initiate a check of the latest revision and then build if the revision is changed).
```

Reviewer: @baymac  Can you accept this suggestion?  Then we can merge this.
